[Bug] master won't compile anymore

**Platform (please complete the following information):**
 - OS and version: Arch Linux (latest)
 - Desktop Environment: Plasma 5.19

**ghostwriter installation (please complete the following information):**
 - ghostwriter version: `master`
 - Installation type: `ghostwriter-git` from the AUR

**Describe the bug**
`master` does not compile with gcc anymore since 63f1f27. 
```
/usr/bin/ld: build/release/dictionary_provider_hunspell.o: in function `(anonymous namespace)::DictionaryHunspell::removeFromSession(QStringLt const&)':
dictionary_provider_hunspell.cpp:(.text+0xc37): undefined reference to `Hunspell::remove(char const*)'
/usr/bin/ld: build/release/dictionary_provider_hunspell.o: in function `(anonymous namespace)::DictionaryHunspell::addToSession(QStringList cst&)':
dictionary_provider_hunspell.cpp:(.text+0xd27): undefined reference to `Hunspell::add(char const*)'
collect2: error: ld returned 1 exit status
make: *** [Makefile:754: build/release/ghostwriter] Error 1
```

**To Reproduce**
Steps to reproduce the behavior:
1. Try to compile `master` with gcc.

**Expected behavior**
It compiles and works as expected.

--- a/src/spelling/dictionary_provider_hunspell.cpp
+++ b/src/spelling/dictionary_provider_hunspell.cpp
@@ -31,7 +31,11 @@
 #include <QStringList>
 #include <QTextCodec>
 
+#ifdef _WIN32
 #include "hunspell/hunspell.hxx"
+#else
+#include <hunspell.hxx>
+#endif
 
 //-----------------------------------------------------------------------------
 
@@ -275,7 +279,11 @@ void DictionaryHunspell::addToPersonal(const QString& word)
 void DictionaryHunspell::addToSession(const QStringList& words)
 {
 	foreach (const QString& word, words) {
+#ifdef _WIN32
 		m_dictionary->add(m_codec->fromUnicode(word).constData());
+#else
+		m_dictionary->add(m_codec->fromUnicode(word).toStdString());
+#endif
 	}
 }
 
@@ -284,7 +292,12 @@ void DictionaryHunspell::addToSession(const QStringList& words)
 void DictionaryHunspell::removeFromSession(const QStringList& words)
 {
 	foreach (const QString& word, words) {
+#ifdef _WIN32
 		m_dictionary->remove(m_codec->fromUnicode(word).constData());
+#else
+		m_dictionary->remove(m_codec->fromUnicode(word).toStdString());
+#endif
+
 	}
 }
 

